### PR TITLE
remove installed from resolve

### PR DESCRIPTION
**Pull request by kalefranz**
If this passes it'll help decouple resolve from extra information in index records just a little bit.

### CODE REVIEW

**Comment by kalefranz** [comment]
@mcg1969 Guessing you have no objection?  It just disentangles some needlessly coupled logic.

File: conda/core/index.py [hunk 1]
```diff
@@ -56,14 +56,7 @@ def supplement_index_with_prefix(index, prefix, channels):
...
     maxp = len(channels) + 1
     for dist, info in iteritems(linked_data(prefix)):
-        if dist in index:
-            # The downloaded repodata takes priority, so we do not overwrite.
-            # We do, however, copy the link information so that the solver
-            # knows this package is installed.
-            old_record = index[dist]
-            link = info.get('link') or EMPTY_LINK
-            index[dist] = IndexRecord.from_objects(old_record, link=link)
-        else:
```
Comment (kalefranz) [inline conda/core/index.py]: we don't need this anymore if we get rid of the `installed` property on resolve, and then use `linked_data` appropriately

File: conda/models/index_record.py [hunk 2]
```diff
@@ -81,8 +61,10 @@ class IndexRecord(DictSafeMixin, ImmutableEntity):  # rename to IndexRecord
...
-    files = ListField(string_types, default=(), required=False)
-    link = ComposableField(Link, required=False)
-
     with_features_depends = MapField(required=False)
     preferred_env = StringField(default=None, required=False, nullable=True)
+
+
+class LinkedPackageRecord(IndexRecord):
+    files = ListField(string_types, default=(), required=False)
+    link = ComposableField(Link, required=False)
```
Comment (kalefranz) [inline conda/models/index_record.py]: To preserved the behavior for what `linked_data` returns, I've finally sub-classed `IndexRecord` and now moved these two fields.  I've had a `TODO` for this for a long time.

**Comment by mcg1969** [comment]
Not so fast. The Resolve object needs to know which packages in the index are installed. Definitely do not merge this until you and I have a chance to talk. It might not be a problem but I just don't know for sure

**Comment by mcg1969** [comment]
I'll rebase this one if you give me permission

**Comment by kalefranz** [comment]
> I would rather we delay this until after we're satisfied that the short-term performance fixes are done.

This isn't something that should go into 4.3.x.  Agreed.

**Comment by kalefranz** [comment]
incorporated into #4474

**Pull request closed without merging**